### PR TITLE
Minor Mountain Time Refactoring (C++) 

### PR DESCRIPTION
**Pull request by ryanchou-dev**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Review by TheGamingMousse — APPROVED** [review]
strong

im assuming it still ACs

**Comment by ryanchou-dev** [comment]
https://csacademy.com/submission/4742678

File: solutions/gold/csa-MountainTime.mdx [hunk 1]
```diff
@@ -78,23 +77,19 @@ struct DSU {
 
 		if (grid[y] > grid[x]) {
 			// we have the answer for the smaller component
-			for (int &i : comp[x]) {
-				ans[i] = intermediate > grid[i] ? 0 : grid[i] - intermediate;
-			}
+			for (int &i : comp[x]) { ans[i] = grid[i] - intermediate; }
```
Comment (SansPapyrus683) [inline solutions/gold/csa-MountainTime.mdx:80]: no yeah now i'm a bit concerned if this assumes operations are performed in a certain order

Comment (bqi343) [inline solutions/gold/csa-MountainTime.mdx:80]:
```suggestion
			for (int i : comp[x]) { ans[i] = grid[i] - intermediate; }
```

Comment (ryanchou-dev) [reply solutions/gold/csa-MountainTime.mdx:80]: hmm well all of component y's elements will be greater than component x's - and intermediate (the element we just added which is the lowest) connects them.

i can add a comment if it's still unclear though ig

Comment (SansPapyrus683) [reply solutions/gold/csa-MountainTime.mdx]: comment would be nice

**Review by bqi343 — APPROVED** [review]

File: solutions/gold/csa-MountainTime.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/gold/csa-MountainTime.mdx]: okay sure

**Pull request merged**